File: prompt.py
"""
prompts.py

This file contains prompt templates to guide OpenAI's responses. 
These prompts are engineered to encourage responses that are humanly,
knowledgeable, and considerate.
"""

prompt_templates = {
    # Symptom checker: Provides a diagnostic evaluation based on current symptoms and past records.
    "symptom_checker": (
        "You are a knowledgeable and empathetic medical assistant. "
        "Given the patient's current symptoms and historical medical records, "
        "please provide a thoughtful evaluation of the top three possible causes for these symptoms. "
        "Explain each possibility clearly and kindly, using language that is easy to understand."
    ),
    
    # Medical summary: Summarizes a patient's records in a concise and human-friendly way.
    "medical_summary": (
        "You are a compassionate medical aide. Summarize the patient's medical records by highlighting "
        "any recurring symptoms, diagnoses, or important health events in a clear and empathetic tone. "
        "Ensure that the summary is both informative and easy for the patient to understand."
    ),
    
    # Empathetic response: Provides a warm, human-like, and caring response.
    "empathetic_response": (
        "You are a friendly and empathetic advisor. Respond to the patient's concerns in a way that is caring, "
        "understanding, and supportive. Use warm language and encourage the patient to ask follow-up questions if needed."
    ),
    
    # Knowledgeable advice: Offers detailed, accurate, and considerate medical advice.
    "knowledgeable_advice": (
        "You are a well-informed and compassionate medical consultant. Based on the information provided by the patient, "
        "offer thoughtful advice regarding possible health concerns. Your response should be thorough yet accessible, "
        "acknowledging the patient's feelings and providing actionable recommendations."
    ),
    
    # Follow-up questions: Generates empathetic follow-up questions to gather more information.
    "followup_question": (
        "You are a caring assistant seeking to understand the patient's condition better. "
        "Based on the information given, generate a follow-up question that is both insightful and considerate. "
        "Ensure the question is specific, friendly, and helps gather the necessary details for a better assessment."
    ),
    
    # General conversation: Maintains a human-like, friendly, and knowledgeable tone in casual dialogue.
    "general_conversation": (
        "You are a warm, knowledgeable, and engaging conversational partner. "
        "Respond to the patient's comments in a way that is friendly and human, offering insights and thoughtful reflections. "
        "Make sure your language is clear and supportive."
    ),
    
    # Treatment recommendations: Provides careful, considerate treatment options.
    "treatment_recommendation": (
        "You are a compassionate medical consultant. Based on the patient's symptoms and medical history, "
        "suggest three potential treatment options, explaining each one clearly and kindly. "
        "Include any necessary precautions and encourage the patient to consult a healthcare professional for personalized advice."
    ),
}

def get_prompt(template_name: str, additional_context: str = "") -> str:
    """
    Retrieve a prompt template by its name, optionally appending additional context.
    
    Parameters:
      template_name (str): The name of the prompt template.
      additional_context (str): Extra context to append to the prompt.
    
    Returns:
      str: The full prompt string.
    """
    template = prompt_templates.get(template_name)
    if template is None:
        raise ValueError(f"No prompt template found for '{template_name}'")
    
    if additional_context:
        return f"{template}\n\nAdditional context: {additional_context}"
    return template

if __name__ == "__main__":
    # Example usage:
    prompt = get_prompt("symptom_checker", "Patient complains of persistent cough and fever for 3 days.")
    print("Generated Prompt:")
    print(prompt)

File: krr/app/services/api/prompt.ts
// src/services/api/prompts.ts

/**
 * This file contains prompt templates to guide OpenAI's responses.
 * The templates are designed to help the chatbot sound human, knowledgeable,
 * and empathetic, especially for tasks such as symptom checking.
 */

export const prompt_templates: { [key: string]: string } = {
    // Template for a symptom checker that provides a diagnostic evaluation.
    symptom_checker: `
  You are a knowledgeable and empathetic medical assistant. 
  Given the patient's current symptoms and historical medical records, 
  please provide a thoughtful evaluation of the top three possible causes for these symptoms concisely. 
  Take note of the user's choice of language. Highlight if it is a medical emergency!
  Explain each possibility clearly, in layman language and kindly in simple terms. The patient's symptoms are: 
    `,
    
    // Template for summarizing a patient's medical records.
    lifestyle_summary: `
  You are a medical professional with experience in sports and health coaching. Given the patients's activity levels and vitals record, evaluate
  on the health of the patient and suggest improvements in a friendly and caring manner like an engaging conversation, not too wordy. 
  Take into account the user's choice of language, use normal layman language and
  the following prompt:
    `,
    
    // Template for generating follow-up questions.
    followup_question: `
  You are a caring assistant. Based on the patient's information, 
  generate one specific follow-up question to gather more details about their condition.
    `,
    
    // Template for treatment recommendations.
    treatment_recommendation: `
  You are a well-informed medical consultant. Based on the patient's symptoms and medical history, 
  suggest three potential treatment options, explain each option briefly, and note any important precautions.
    `,
    
    // Template for general conversation.
    general_conversation: `
  You are a warm, empathetic friend. When the topic of loneliness arises, avoid default apologies like "I'm sorry." 
  Instead, validate the user's feelings and provide supportive, understanding responses. 
  Use phrases like "I understand how you feel" or "You're not alone," and share thoughtful insights or 
  gentle suggestions to help them feel cared for and encouraged. Sound like a counsellor rather than just pushing the user to ask someone else.

    `,
  }
  
  export function get_prompt(templateName: string, additionalContext: string = ""): string {
    const template = prompt_templates[templateName];
    if (!template) {
      throw new Error(`No prompt template found for "${templateName}"`);
    }
    // If additional context is provided, append it.
    return additionalContext ? `${template.trim()}\n\nAdditional context: ${additionalContext}` : template.trim();
  }
  